src/main/java/com/google/sps/servlets/ListOrganizationsServlet.java

```diff
@@ -96,9 +73,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(gson.toJson(requestedOrganizations));
   }
 
-  public Query ConstructQueryForUserInfo(String userID) {
-    Query query = new Query("Distributor").setFilter(new FilterPredicate("moderatorList",
-                    FilterOperator.EQUAL, userID)).addSort("creationTimeStamp", SortDirection.DESCENDING);
+  /* Given a user id, this function checks if that user has a maintainer role in the datastore */
+  public boolean userIsMaintainer(String userId) {
+    //TODO(): Check if given user ID has maintainer role in the datastore
+    return true;
+  }
+
+  /* This function constructs a query based on the request parameters & user's role */
+  public Query getQueryFromParams(HttpServletRequest request, UserService userService) {
+    Query query = new Query("Distributor").addSort("creationTimeStamp", SortDirection.DESCENDING);;
+
+    /* displayUserOrgsParameter is true when user only wants to see orgs they moderate*/
```

Comment:
Can you extract a function and create a new class:

RequestParameterHelper {
  bool getParameterAsBooleanOrDefault(String param, bool default) {
     String value = request.getParam(..);
     if (value == null) return default;
     // return if true.
  }
}